Marlin/src/lcd/menu/menu_motion.cpp

```diff
@@ -417,10 +416,9 @@ void menu_move() {
   void menu_ft_motion() {
     // Define stuff ahead of the menu loop
     MString<20> shaper_name[NUM_AXES_SHAPED] {};
-    #if HAS_X_AXIS
-      for (uint_fast8_t a = X_AXIS; a < NUM_AXES_SHAPED; ++a)
-        shaper_name[a] = get_shaper_name(AxisEnum(a));
-    #endif
+    TERN0(HAS_X_AXIS,shaper_name[X_AXIS] = get_shaper_name(X_AXIS);)
+    TERN0(HAS_Y_AXIS,shaper_name[Y_AXIS] = get_shaper_name(Y_AXIS);)
+    
```

Comment:
You would want to use `TERN_` here, not `TERN0`. And, this isn't a good idea. It's better to leave it as a loop — one less thing we need to change if Z shaping becomes a thing. And there will not be any case where there is a Y axis without an X axis.